libcnb-package/src/lib.rs

```diff
@@ -131,7 +167,11 @@ fn create_file_symlink<P: AsRef<Path>, Q: AsRef<Path>>(
 /// This function ensures the resulting name is valid and does not contain problematic characters
 /// such as `/`.
 pub fn default_buildpack_directory_name<BM>(
-    buildpack_descriptor: &SingleBuildpackDescriptor<BM>,
+    buildpack_descriptor: &BuildpackDescriptor<BM>,
 ) -> String {
-    buildpack_descriptor.buildpack.id.replace('/', "_")
+    let id = match buildpack_descriptor {
+        BuildpackDescriptor::Single(descriptor) => &descriptor.buildpack.id,
+        BuildpackDescriptor::Meta(descriptor) => &descriptor.buildpack.id,
+    };
+    id.replace('/', "_")
```

Comment:
Nit: We could move the logic of extracting the buildpack data from a buildpack descriptor into its own function and make it available for everyone. This would also benefit one test in this PR that uses the exact same code. I'm thinking:

```rust
buildpack_descriptor.buildpack().id.replace('/', "_")
```

here and the following in `libcnb-data/src/buildpack/mod.rs`:

```rust
impl<BM> BuildpackDescriptor<BM> {
    pub fn buildpack(&self) -> &Buildpack {
        match self {
            BuildpackDescriptor::Single(descriptor) => &descriptor.buildpack
            BuildpackDescriptor::Meta(descriptor) => &descriptor.buildpack
        }
    }
}
```